### PR TITLE
Implement measure_text_width with no allocation

### PR DESCRIPTION
**Pull request by remi-dupre**
I implemented that while working on optimizing something I don't remember, it might still be useful.

Also, I noticed that the test for this function was wrong when only using the `unicode-width` feature (probably due to some update in the corresponding?).

```bash
# Running on origin/main
$ cargo test --no-default-features -F unicode-width 
running 6 tests
test utils::test_pad_str ... ok
test utils::test_truncate_str_no_ansi ... ok
test utils::test_pad_str_with ... ok
test utils::test_text_width ... FAILED
test utils::test_attributes_many ... ok
test utils::test_attributes_single ... ok

failures:

---- utils::test_text_width stdout ----
thread 'utils::test_text_width' panicked at 'assertion failed: `(left == right)`
  left: `21`,
 right: `17`', src/utils.rs:940:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I've completed the Makefile so that all possible sets of feature are tested, as the project is not too big I don't think it's too heavy.

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -71,7 +66,18 @@ pub fn set_colors_enabled_stderr(val: bool) {
 
 /// Measure the width of a string in terminal characters.
 pub fn measure_text_width(s: &str) -> usize {
-    str_width(&strip_ansi_codes(s))
+    dbg!(&s);
+    #[cfg(feature = "ansi-parsing")]
+    {
+        AnsiCodeIterator::new(s)
+            .filter(|(_, is_ansi)| !is_ansi)
+            .map(|(sub, _)| str_width(sub))
```
[the remark this answers is not in the document]
Comment (remi-dupre) [reply src/utils.rs:74]: I find it less readable, but that's your project 👍

[the remark this answers is not in the document]
Comment (remi-dupre) [reply src/utils.rs:74]: Actuallyyyy, clippy doesn't like it either! 🤓 

https://rust-lang.github.io/rust-clippy/master/index.html#filter_map_bool_then

**Comment by remi-dupre** [comment]
I'll investigate on why the tests fails later, I start to realy suspects some behavior change in unicode_width (I know this sounds cliché but it worked on my computer 😢)

File: src/utils.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (djc) [reply src/utils.rs:73]: Fair enough. In that case, I'd like it to be written like this (assuming clippy is okay with it):

```rust
.filter_map(|(s, is_ansi)| match is_ansi {
    false => Some(str_width(s)),
    true => None,
})
```

[the remark this answers is not in the document]
Comment (remi-dupre) [reply src/utils.rs]: Clippy is okay with that :+1:

**Comment by remi-dupre** [comment]
Tests are fixed.

This was just a rooky mistake of forgetting to put `.force_styling(true)` in the new test case, it took me way too much time to figure that out :sweat_smile:

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
Nice, thanks!
